gdk/commands/component/PublishCommand.py

```diff
@@ -339,7 +339,7 @@ def update_and_create_recipe_file(self, component_name, component_version):
                     logging.debug("Updating artifact URI of '{}' in the recipe file.".format(artifact_file))
                     artifact["URI"] = f"{artifact_uri}/{artifact_file}"
                 else:
-                    raise Exception(
+                    logging.warning(
                         f"Could not find the artifact file specified in the recipe '{artifact_file}' inside the build folder"
                         f" '{gg_build_component_artifacts}'."
                     )
```

Comment:
Can we verify if the file is on S3 or not? If its not on S3 then it should be an exception.

Reply:
That check is performed during gdk `build` and will again be with checked by GG's create component version api after the recipe is updated. It is redundant to perform this same check again here. So, just warned saying that it's not in the build folder. 

Check in build: https://github.com/aws-greengrass/aws-greengrass-gdk-cli/blob/main/gdk/commands/component/BuildCommand.py#L370-L384

Create component after updating recipes: https://github.com/aws-greengrass/aws-greengrass-gdk-cli/blob/main/gdk/commands/component/PublishCommand.py#L43-L47